File: Drift/Event.m
#import "Event.h"

@interface Event ()
@end

@implementation Event

- (void)awakeFromInsert {
    [super awakeFromInsert];

    self.guid = [[NSUUID UUID] UUIDString];
}

- (BOOL)isActive {
    return self.stopDate == nil;
}

@end

File: Drift/Event.swift
import Foundation
import CoreData

final class Event: NSManagedObject {
    convenience init(_ context: NSManagedObjectContext,
                startDate: Date,
                guid: String? = UUID().uuidString,
                stopDate: Date? = nil,
                inTag: Tag? = nil)
    {
        guard let entity = NSEntityDescription.entity(forEntityName: self.dynamicType.entityName, in: context) else {
            fatalError("Unable to create entity for Event")
        }
        
        self.init(entity: entity, insertInto: context)
        
        self.startDate = startDate
        self.guid = guid
        self.stopDate = stopDate
        self.inTag = inTag
    }
}
